packages/flutter_tools/test/general.shard/android/deferred_components_setup_validator_test.dart

```diff
@@ -0,0 +1,852 @@
+import 'package:flutter_tools/src/base/deferred_component.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/build_system/build_system.dart';
+import 'package:flutter_tools/src/build_system/targets/common.dart';
+import 'package:flutter_tools/src/globals.dart' as globals;
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+
+void main() {
+  FileSystem fileSystem;
+  BufferLogger logger;
+  Environment env;
+
+  Environment createEnvironment() {
+    final Map<String, String> defines = <String, String>{ kSplitAot: 'true' };
+    final Environment result = Environment(
+      outputDir: fileSystem.directory('/output'),
+      buildDir: fileSystem.directory('/build'),
+      projectDir: fileSystem.directory('/project'),
+      defines: defines,
+      inputs: <String, String>{},
+      cacheDir: fileSystem.directory('/cache'),
+      flutterRootDir: fileSystem.directory('/flutter_root'),
+      artifacts: globals.artifacts,
+      fileSystem: fileSystem,
+      logger: logger,
+      processManager: globals.processManager,
+      engineVersion: 'invalidEngineVersion',
+    );
+    return result;
+  }
+
+  setUp(() {
+    fileSystem = MemoryFileSystem.test();
+    logger = BufferLogger.test();
+    env = createEnvironment();
+  });
+
+  test('No checks passes', () async {
+    final DeferredComponentsSetupValidator validator = DeferredComponentsSetupValidator(
+      env,
+      exitOnFail: false,
+      title: 'test check',
+    );
+    validator.displayResults();
+    validator.attemptToolExit();
+    expect(logger.statusText, 'test check passed.\n');
+  });
+
+  test('clearTempDir passes', () async {
+    final DeferredComponentsSetupValidator validator = DeferredComponentsSetupValidator(
+      env,
+      exitOnFail: false,
+      title: 'test check',
+    );
+    validator.displayResults();
+    validator.attemptToolExit();
+
+    expect(logger.statusText, 'test check passed.\n');
+  });
+
+  test('writeGolden passes', () async {
+    final File goldenFile = env.projectDir.childFile(DeferredComponentsSetupValidator.kDeferredComponentsGoldenFileName);
+    if (goldenFile.existsSync()) {
+      goldenFile.deleteSync();
+    }
+    final DeferredComponentsSetupValidator validator = DeferredComponentsSetupValidator(
+      env,
+      exitOnFail: false,
+      title: 'test check',
+    );
+    validator.writeGolden(
+      <LoadingUnit>[
+        LoadingUnit(id: 2, libraries: <String>['lib1']),
+        LoadingUnit(id: 3, libraries: <String>['lib2', 'lib3']),
+      ],
+    );
+    validator.displayResults();
+    validator.attemptToolExit();
+
+    expect(logger.statusText, 'test check passed.\n');
+
+    final File expectedFile = env.projectDir.childFile('deferred_components_golden.yaml');
+
+    expect(expectedFile.existsSync(), true);
+    const String expectedContents =
+'''
+loading-units:
```

Comment:
If this yaml can be provided by users by hand, then there should be tests that the tool does an appropriate `throwToolExit` when it is malformed.